src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -117,16 +117,38 @@ public async Task<IUser> CreateUserAsync(IUser user, string password, Action<str
             throw new ArgumentException("Expected a User instance.", nameof(user));
         }
 
+        var hasPassword = !string.IsNullOrWhiteSpace(password);
+
         // Accounts can be created with no password.
-        var identityResult = string.IsNullOrWhiteSpace(password)
-            ? await _userManager.CreateAsync(user)
-            : await _userManager.CreateAsync(user, password);
+        var identityResult = hasPassword
+            ? await _userManager.CreateAsync(user, password)
+            : await _userManager.CreateAsync(user);
+
         if (!identityResult.Succeeded)
         {
+            if (hasPassword)
+            {
+                _logger.LogInformation(3, "Unable to create a new account with password.");
```

Comment:
If you use the event ID parameter then it should be unique. I don't think it's particularly useful, though.

Reply:
Agreed that it not useful. It used like this is many places so I left it as is.

Reply:
I removed the useless event ids